src/Illuminate/Http/Client/PendingRequest.php

```diff
@@ -385,6 +392,19 @@ public function withOptions(array $options)
         });
     }
 
+    /**
+     * Add new middleware the client handlerstack.
+     *
+     * @param  callable  $middleware
+     * @return $this
+     */
+    public function withMiddleware(callable $middleware)
+    {
+        return tap($this, function ($request) use ($middleware) {
+            return $this->middleware = collect($this->middleware)->push($middleware);
```

Comment:
`return $this->middleware ...` wouldn't have any effect for a `tap()` callback since the return value is ignored.

Either way, this method looks much simpler by skipping `tap()`:

```php
$this->middleware = collect($this->middleware)->push($middleware);

return $this;
```